src/main.cpp

```diff
@@ -658,14 +658,8 @@ bool IsStandardTx(const CTransaction& tx, string& reason)
 
 bool IsFinalTx(const CTransaction &tx, int nBlockHeight, int64_t nBlockTime)
 {
-    AssertLockHeld(cs_main);
-    // Time based nLockTime implemented in 0.1.6
     if (tx.nLockTime == 0)
         return true;
-    if (nBlockHeight == 0)
```

Comment:
Are you sure none of the current callers of IsFinalTx pass nBlockHeight or nBlockTime = 0 ?
Seriously, it seems much safer to keep this function as it is and call the "new"/consensus-friendly one from that one.
If you're breaking the old one, why keep to of them at all?
The only reason I maintain the old one in #6063 is to make sure it keeps being functionally identical. If you can't warranty that, I would just use the consensus-friendly version everywhere.